src/bin/part1.rs

```diff
@@ -99,12 +112,7 @@ impl SantaTracker {
     }
 
     pub fn perform_move(&mut self, direction: Direction) {
-        let new_position = match direction {
-            Direction::North => Pos::new(self.current_position.x, self.current_position.y + 1),
-            Direction::South => Pos::new(self.current_position.x, self.current_position.y - 1),
-            Direction::East => Pos::new(self.current_position.x + 1, self.current_position.y),
-            Direction::West => Pos::new(self.current_position.x - 1, self.current_position.y),
-        };
+        let new_position = self.current_position + direction;
```

Comment:
Here we get top replace the old `match` statement that was here with a simple addition statement, moving the actual move logic into our `Add` implementation.